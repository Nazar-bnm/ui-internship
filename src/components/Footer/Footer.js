import React from 'react';

const Footer = () => (
  <div>
    <h2>Footer</h2>
  </div>
);

export default Footer;
