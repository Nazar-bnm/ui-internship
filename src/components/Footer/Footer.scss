@import '../../scss/global_styles';

.footer {
  background-color: $black;
  padding-top: 30px;

  a {
    text-decoration: none;
  }

  &__link {
    margin: 0 3px;
    text-decoration: underline;

    &:hover {
      color: $white;
    }
  }

  &__title {
    @include footer-title;
  }

  &__text {
    @include footer-text;
  }

  &__section {
    padding: 10px 5px 10px 0;

    h4:hover {
      color: $white;
      text-decoration: underline;
    }
  }

  &__image {
    width: 100%;
    height: auto;
    margin-top: 5px;
  }

  &__info {
    @include flex($align: center);
    padding: 10px 0;

    .icon.big {
      color: $emperor;
      margin: 0 5px;

      &:hover {
        color: $white;
      }
    }
  }

  &__connect {
    @include flex($justify: space-between);
    border-top: 1px solid $white;
    border-bottom: 1px solid $white;
    
    .footer__title {
      margin: 0;
    }
  }

  &-tablet {
    display: none;
  }
}

@include media-for(tablet) {
  .footer {
    padding: 0;

    .container {
      display: none;
    }

    &__desktop {
      display: none;
    }

    &__info {
      @include flex($wrap: wrap, $justify: center);
    }

    &-tablet {
      @include flex($justify: center);

      &__link { 
        @include flex($align: center);
        height: 11px;
        padding: 0 10px;
        border-right: 1px solid $white;
        margin: 10px 0;

        &:last-child {
          border: 1px solid $black;
        }

        h4 {
          line-height: 0;
        }
      }
    }
  }
}
