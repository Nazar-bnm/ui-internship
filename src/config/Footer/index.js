export const linkListItems = {
  guide: {
    heading: 'Shopping guide',
    data: [
      { title: 'About G-Star Raw', link: '#' },
      { title: 'Corporate Responsibility', link: '#' },
      { title: 'Press Room', link: '#' },
      { title: 'Careers', link: '#' },
      { title: 'G-Star Retailers', link: '#' },
      { title: 'Open a G-Star Store', link: '#' },
    ],
  },
  help: {
    heading: 'Help & Info',
    data: [
      { title: 'FAQ', link: '#' },
      { title: 'Contact', link: '#' },
      { title: 'Privacy Policy', link: '#' },
      { title: 'Terms & Conditions', link: '#' },
      { title: 'Disclaimer', link: '#' },
    ],
  },
};

export const icons = [
  { name: 'twitter', link: 'https://twitter.com/griddynamics' },
  { name: 'mail', link: 'mailto:khrystynavaskovych@gmail.com' },
  { name: 'facebook', link: 'https://www.facebook.com/griddynamics' },
  { name: 'linkedin', link: 'https://www.linkedin.com/company/grid-dynamics/' },
  { name: 'instagram', link: 'https://www.instagram.com/griddynamics_ua/' },
  {
    name: 'google plus g',
    link: 'https://plus.google.com/115302417170674279390',
  },
];

export const paymentIcons = [
  { name: 'cc paypal', link: '#' },
  { name: 'cc visa', link: '#' },
  { name: 'cc mastercard', link: '#' },
];

export const linkListItemsTablet = [
  { title: 'About G-Star Raw', link: '#' },
  { title: 'G-Star Retailers', link: '#' },
  { title: 'FAQ', link: '#' },
  { title: 'Contact', link: '#' },
  { title: 'Privacy Policy', link: '#' },
  { title: 'Terms & Conditions', link: '#' },
];
