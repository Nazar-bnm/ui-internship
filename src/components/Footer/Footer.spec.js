import React from 'react';
import { shallow } from 'enzyme';
import Footer from './Footer';

describe('<Footer />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Footer />);
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
