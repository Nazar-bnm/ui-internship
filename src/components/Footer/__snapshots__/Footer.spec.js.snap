// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
<div
  className="footer"
>
  <div
    className="container content footer__desktop"
  >
    <div
      className="col-9"
    >
      <div
        className="container"
      >
        <div
          className="footer__section col-3"
        >
          <Linklist
            links={
              Object {
                "data": Array [
                  Object {
                    "link": "#",
                    "title": "About G-Star Raw",
                  },
                  Object {
                    "link": "#",
                    "title": "Corporate Responsibility",
                  },
                  Object {
                    "link": "#",
                    "title": "Press Room",
                  },
                  Object {
                    "link": "#",
                    "title": "Careers",
                  },
                  Object {
                    "link": "#",
                    "title": "G-Star Retailers",
                  },
                  Object {
                    "link": "#",
                    "title": "Open a G-Star Store",
                  },
                ],
                "heading": "Shopping guide",
              }
            }
          />
        </div>
        <div
          className="footer__section col-3"
        >
          <Linklist
            links={
              Object {
                "data": Array [
                  Object {
                    "link": "#",
                    "title": "FAQ",
                  },
                  Object {
                    "link": "#",
                    "title": "Contact",
                  },
                  Object {
                    "link": "#",
                    "title": "Privacy Policy",
                  },
                  Object {
                    "link": "#",
                    "title": "Terms & Conditions",
                  },
                  Object {
                    "link": "#",
                    "title": "Disclaimer",
                  },
                ],
                "heading": "Help & Info",
              }
            }
          />
        </div>
        <div
          className="footer__section col-6"
        >
          <h3
            className="footer__title"
          >
            Twitter
          </h3>
          <p
            className="footer__text"
          >
            check out the new upcoming phone !! #Valentine’s Day #Australia
            <a
              className="footer__text footer__link"
              href="http://t.co/StOrIs63bd"
            >
              http://t.co/wRye6rBElN
            </a>
             - 7 hours ago 
          </p>
          <p
            className="footer__text"
          >
            your solution partner:
            <a
              className="footer__text footer__link"
              href="http://t.co/StOrIs63bd"
            >
              http://t.co/StOrIs63bd
            </a>
            #IT #websitedevlopment - 8 hours ago
          </p>
        </div>
      </div>
      <div
        className="container"
      >
        <div
          className="footer__section col-5"
        >
          <h3
            className="footer__title"
          >
            About us
          </h3>
          <p
            className="footer__text"
          >
            One could refuse to pay expensive translators, the European languages are members of the same family. Their separate existence is a myth. Uses read more
          </p>
        </div>
        <div
          className="footer__section col-6 col-right"
        >
          <div
            className="container"
          >
            <div
              className="col-5"
            >
              <img
                alt="google map"
                className="footer__image"
                src="src/assets/img/content/map.png"
              />
            </div>
            <div
              className="col-6 col-center"
            >
              <h3
                className="footer__title"
              >
                Store location
              </h3>
              <p
                className="footer__text"
              >
                Company ltd.co 234 Fake address name, Fake city name, Country 01234 (000) 123 456 xxx
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="footer__section col-3"
    >
      <h3
        className="footer__title"
      >
        Facebook
      </h3>
      <img
        className="footer__image"
        src="src/assets/img/content/facebook.png"
      />
    </div>
  </div>
  <ul
    className="footer__section content footer-tablet"
  >
    <li
      className="footer-tablet__link"
      key="About G-Star Raw"
    >
      <a
        href="#"
      >
        <h4
          className="footer__text"
        >
          About G-Star Raw
        </h4>
      </a>
    </li>
    <li
      className="footer-tablet__link"
      key="G-Star Retailers"
    >
      <a
        href="#"
      >
        <h4
          className="footer__text"
        >
          G-Star Retailers
        </h4>
      </a>
    </li>
    <li
      className="footer-tablet__link"
      key="FAQ"
    >
      <a
        href="#"
      >
        <h4
          className="footer__text"
        >
          FAQ
        </h4>
      </a>
    </li>
    <li
      className="footer-tablet__link"
      key="Contact"
    >
      <a
        href="#"
      >
        <h4
          className="footer__text"
        >
          Contact
        </h4>
      </a>
    </li>
    <li
      className="footer-tablet__link"
      key="Privacy Policy"
    >
      <a
        href="#"
      >
        <h4
          className="footer__text"
        >
          Privacy Policy
        </h4>
      </a>
    </li>
    <li
      className="footer-tablet__link"
      key="Terms & Conditions"
    >
      <a
        href="#"
      >
        <h4
          className="footer__text"
        >
          Terms & Conditions
        </h4>
      </a>
    </li>
  </ul>
  <div
    className="content footer__info footer__connect"
  >
    <JoinUs />
    <ContactDetails />
  </div>
  <div
    className="content footer__info"
  >
    <div
      className="col-6"
    >
      <h4
        className="footer__text"
      >
        © 2016 The shop. All rights reserved.
      </h4>
    </div>
    <div
      className="col-right"
    >
      <ul
        className="footer__payment"
      >
        <li
          key="cc paypal"
        >
           
          <a
            href="#"
          >
            <i
              className="cc paypal icon big"
            />
          </a>
           
        </li>
        <li
          key="cc visa"
        >
           
          <a
            href="#"
          >
            <i
              className="cc visa icon big"
            />
          </a>
           
        </li>
        <li
          key="cc mastercard"
        >
           
          <a
            href="#"
          >
            <i
              className="cc mastercard icon big"
            />
          </a>
           
        </li>
      </ul>
    </div>
  </div>
</div>
`;
